docs/Training-Self-Play.md

```diff
@@ -0,0 +1,51 @@
+# Training with Self-Play
+
+ML-Agents provides the functionality to train symmetric, adversarial games with [Self-Play](https://openai.com/blog/competitive-self-play/).
```

Comment:
I think we should tell the reader (game-dev) what symmetric implies about their game and the observation/action space